### ISSUE TITLE
Suggestion: Create js-runtime agnostic utils for cli developer tools 

### ISSUE THREAD

**Issue by guy-borderless** [denoland/deno#20799]
Due to issue #19826, I'm writing this to address what seems to be a common problem with tools incompatible with deno. A lot of CLIs developed by dev-tools, frameworks, and dev-management teams that are aimed at DX functionalities like init, migrations, testing, upgrades, debug environment setup and so forth assume the existence of a package.json. 

The fact that a very trendy tool like drizzle for so long has one of its main value propositions not working on deno because of that is a problem big enough that it may turn people away from the deno runtime in general. I'm working on patching the dev cli of the wonderful [trigger.dev](https://github.com/triggerdotdev/trigger.dev/pull/531) where (unlike the case with drizzle-kit) the cli is a must to effectively use the platform.  

The trigger libraries, including their framework integrations, like the drizzle libs, all work with the npm specifier.  Just the assumption of a package.json holds off deno compatibility in trigger.dev for quite a long time. unlike trigger drizzle-kit is not open-source so it can not be patched by the community. 

I think this should probably be addressed in a more general way. One idea could be to provide cli developers some form of convenient encapsulation over runtime config. This way it is easy to patch tools (and also develop them runtime agnostic from the get-go) since you won't need to open the deno docs (and potentially other js runtimes' docs if those would also be supported). 

something like:
```

1. `runtime.installedPackages(projectRoot): {packageName, repository, version}[]`, taken from package.json / deno.json(c) +import_map)
2. `runtime.installPackage`
3. `runtime.updatePackage`
4. `runtime.`[checkUpdates](https://www.npmjs.com/package/npm-check-updates)(projectRoot)
5. `runtime.userConfig` any user keys in package.json / deno.json
6. `runtime.scripts` returns runtime.scripts (add, replace, run)

```

This I hope can encourage developers to build more cross-runtime dev tools.

**Comment by bartlomieju** [comment]
Getting `drizzle-kit` and `trigger` would definitely be a welcome addition, but I'm not sure I understand what is being proposed in this issue. Are you talking about adding runtime APIs that would return list of installed packages and install packages on demand? How would that help if these tools rely on the presence of `package.json`?

**Comment by guy-borderless** [comment]
> Getting `drizzle-kit` and `trigger` would definitely be a welcome addition, but I'm not sure I understand what is being proposed in this issue. Are you talking about adding runtime APIs that would return list of installed packages and install packages on demand? How would that help if these tools rely on the presence of `package.json`?

No, not to deno itself. Just provide a package with these in deno.land & npm.

**Comment by guy-borderless** [comment]
Just to clarify, the above mentioned dev tooling apis should at least work on both node-js and deno project directories.

**Comment by bartlomieju** [comment]
I still think this is out of scope for Deno, so I'm gonna close this issue for the time being.
